tensorflow_addons/losses/triplet.py

```diff
@@ -79,6 +83,11 @@ def triplet_semihard_loss(
       y_pred: 2-D float `Tensor` of embedding vectors. Embeddings should
         be l2 normalized.
       margin: Float, margin term in the loss definition.
+      squared: bool, determines whether L-1 or L-2 norm is used
+      dist_metric: str or function, determines distance metric:
+                   "L_2" for euclidean distance
```

Comment:
hmmm having bot h squared and "L_2" is confusing, i.e. if people want "L_1" they need to set "L_2" and squared = False, I think adding an "L_1" option to dist_metric would be better than having `squared` argument if we are using dist_metric this way :). What do you think ?

NIT: May you also (quickly) check with TensorFlow repo how they name dist_metric? Do they use `distance_metric` or short-handed `dist_metric` for similar argument? This is for consistency :)

Reply:
I added the option for L-1 distance and removed the squared argument. I changed dist_metric to distance_metric, because even though I wasn't able to find that exact name anywhere else in the repo, the long version seemed more in line with the rest of tensorflow's code. I have also completely overhauled the tests to work with the restructured function.